### PR TITLE
fix: Upgrade terraform-config-inspect

### PR DESCRIPTION
**Pull request by vincenthsh**
### Summary

It seems some public tf modules can not be parsed by current referenced `terraform-config-inspect` version

### Test Plan

following fogg.yml snippet:

```yml
# ...
envs:
  shared-services:
    components:
      atlantis:
        module_source: terraform/modules/atlantis
      state-s3-backend:
        module_name: backend
        module_source: "github.com/nozaq/terraform-aws-remote-state-s3-backend?ref=v1.3.3"
# ...
```

results in:

```sh
fogg apply
INFO .fogg-version templated                      
INFO .gitattributes copied                        
INFO .gitignore copied                            
...
INFO terraform/envs/shared-services/state-s3-backend/Makefile templated 
INFO terraform/envs/shared-services/state-s3-backend/fogg.tf templated 
ERROR: unable to apply envs: unable to apply module invocation: could not download or parse module: There was an issue loading the module: Variables not allowed: Variables may not be used here.
```

### References

Identified by running HEAD of terraform-config-inspect binary against the above module and it worked.
Tested by compiling fogg with the updated `go.mod` and running against the above snippet

### CODE REVIEW

**Review by jakeyheath — APPROVED** [review]

**Comment by jakeyheath** [comment]
Thanks!

**Comment by jakeyheath** [comment]
We might need to update to go 1.18 for this new dep: https://github.com/chanzuckerberg/fogg/actions/runs/3066366823/jobs/4985710584. I don't mind updating if you mind adding that to this PR.

**Comment by vincenthsh** [comment]
oops, yeah I'm on go 1.19, didn't realise this needed go 1.18

**Comment by vincenthsh** [comment]
done @jakeyheath

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
 
       - uses: actions/setup-go@v2
         with:
-          go-version: 1.17
+          go-version: 1.18
```
Comment (jakeyheath) [inline .github/workflows/release.yml:50]: Mind changing this to use:

```suggestion
          go-version-file: 'path/to/go.mod'
```

https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file

**Comment by vincenthsh** [comment]
changed it through the web UI 🙈

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v2
```
Comment (vincenthsh) [inline .github/workflows/build.yml:8]:
```suggestion
      - uses: actions/setup-go@v3
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -17,7 +17,7 @@ jobs:
...
```
Comment (vincenthsh) [inline .github/workflows/build.yml:18]:
```suggestion
      - uses: actions/setup-go@v3
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -47,7 +47,7 @@ jobs:
 
       - uses: actions/setup-go@v2
```
Comment (vincenthsh) [inline .github/workflows/release.yml:48]:
```suggestion
      - uses: actions/setup-go@v3
```

**Review by vincenthsh — COMMENTED** [review]
I guess I forgot to bump the version for the setup-go GitHub Action

**Comment by vincenthsh** [comment]
ready to trigger the CI runs again? @jakeyheath

**Comment by vincenthsh** [comment]
any luck to get this merged today?

**Pull request merged**